rust/docs/parsing.md

```
@@ -0,0 +1,170 @@
+so this lookup means comparing the input character against these intervals.
+Further, at each step, we need to check if the current state is an accepting state.
+In a sense, this DFA loop is an "interpreter" for instructions "goto next state" and "record match".
+
+To speed up matching, we compile DFAs into native functions.
+Currently, Log Surgeon just in time (JIT) compiles them for ease of deployment,
+but conceptually the DFAs are the same as if they were "ahead of time" compiled.
+
+Each state in the DFA will be compiled to a short sequence of basic blocks
+that looks like/roughly corresponds to the following pseudocode:
+
+```
```

markdownlint-cli2 (0.22.1)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)